apps/initializer/lib/initializer.ex

```diff
@@ -49,6 +55,18 @@ defmodule Initializer do
         end
       end
 
+      @retry with: constant_backoff(100) |> take(10)
+      defp do_on_start(state) do
+        case on_start(state) do
+          {:ok, new_state} ->
+            {:ok, new_state}
+
+          {:error, reason} ->
+            {:error,
+             "on_start for application init failed, retrying reading Brook state from store..."}
```

Comment:
If i recall @bbalser's suggestion, if you change `do_on_start/1` to a `with` block and _dont_ worry about an else block, the `{:error, reason}` coming out of the implementing application will automatically bubble up to the case statement in the initializer's handle_continue.

Reply:
Fixed.